### PR TITLE
FieldError shim to avoid breaks in tests

### PR DESCRIPTION
**Pull request by arhik**
Original commit introducing FieldError exception is 7e5d9a3. Since existing packages were using `ErrorException` to capture `no field` errors, we need to have a shim to avoid test breaks for external packages with such uses until they update to `FieldError`. 

addressing `need shim for FieldError` #54807

### CODE REVIEW

File: stdlib/Test/src/Test.jl
Comment (arhik) [inline stdlib/Test/src/Test.jl:809]: May be Union is better.

**Comment by arhik** [comment]
fixes #54807

**Comment by DilumAluthge** [comment]
Can you add a test for this?

**Comment by arhik** [comment]
Sure. Is it possible to check with nanosoldier if this actually solves the `FieldError` problem while I work on tests.

**Comment by arhik** [comment]
I tested it on three packages @vtjnash mentioned. I can confirm two packages `UnPack` and `Parameters` tests pass now. Something else is going on with `FunctionOperators`, so I can't confirm.

**Comment by DilumAluthge** [comment]
@nanosoldier `runtests()`

**Comment by arhik** [comment]
@DilumAluthge Thank you.

File: stdlib/Test/src/Test.jl
Comment (LilithHafner) [inline stdlib/Test/src/Test.jl:815]: It's simpler and less repetitive to add this branch to the inner if-else:
```suggestion
        if isa(extype, Type)
            success =
                if from_macroexpand && extype == LoadError && exc isa Exception
                    Base.depwarn("macroexpand no longer throws a LoadError so `@test_throws LoadError ...` is deprecated and passed without checking the error type!", :do_test_throws)
                    true
                elseif extype == ErrorException && isa(exc, FieldError)
                    Base.depwarn(lazy"ErrorException should no longer be used to test field access; FieldError should be used instead!", :do_test_throws)
                    true
                else
                    isa(exc, extype)
                end
```

**Review by LilithHafner — COMMENTED** [review]
Seems like a reasonable approach. FWIW we explicitly do not guarantee that exceptions will remain consistent across minor versions of Julia.

**Review by LilithHafner — APPROVED** [review]
LGTM, pending PkgEval results.

Thanks!

**Comment by arhik** [comment]
Just checking if I can trigger it.

@nanosoldier `runtests()`

**Comment by nanosoldier** [comment]
The package evaluation job [you requested](https://github.com/JuliaLang/julia/pull/54809#issuecomment-2169180297) has completed - possible new issues were detected.
The [**full report**](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/945545f_vs_d654415/report.html) is available.

**Comment by arhik** [comment]
I triggered pkgevals test out of curiosity. I went through `Nanosoldier.jl` to figure how to cancel it. I couldn't. It takes approx 10 hours to get the reports. If there is a way to stop please help me stop it. If it cannot be stopped. Sorry.

**Comment by LilithHafner** [comment]
Nanosoldier looks good; merging.

In theory, only public committers to this repo can trigger nanosoldier, but we'll see.

**Comment by nanosoldier** [comment]
The package evaluation job [you requested](https://github.com/JuliaLang/julia/pull/54809#issuecomment-2170280120) has completed - possible new issues were detected.
The [**full report**](https://s3.amazonaws.com/julialang-reports/nanosoldier/pkgeval/by_hash/d01ffab_vs_d654415/report.html) is available.

**Pull request merged**
